### PR TITLE
Implement and test get-object-tagging api

### PR DESCRIPTION
**Pull request by priyanka25081999**
- Implement and test s3 get-object-tagging 
- Experiment get-object-tagging and add to experiments folder
- Update Makefile and Requirements file

Signed-off-by: Priyanka Salunke <priyanka.s.salunke@seagate.com>

### CODE REVIEW

File: s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+        self._timer.start()
+        try:
+            async with self._session.get_client_session().get(
+                    self._session.endpoint + request_uri,
+                    params=query_params, headers=headers) as resp:
+
+                self._logger.info(
+                    fmt_reqid_log(
+                        self._request_id) +
```
Comment (kaustubh-d) [inline s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py:101]: move this to previous line

File: s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                self._logger.info(
+                    'Response url {}'.format(
+                        self._session.endpoint + request_uri))
+
+                if resp.status == 200:
+                    self._response_headers = resp.headers
+
+                    total_received = 0
+                    received_tagset = ''
+                    while True:
```
Comment (kaustubh-d) [inline s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py:113]: `received_tagset = await resp.text()`

File: s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py (hunk 1 above)
Comment (priyanka25081999) [reply s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py]: done.

File: s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py (hunk 2 above)
Comment (priyanka25081999) [reply s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py]: done

File: s3/replication/common/tests/system/config/config.yaml [hunk 3]
```diff
@@ -4,6 +4,8 @@ s3_region: us-west2
 source_bucket_name: sourcebucket
 target_bucket_name: targetbucket
 object_name_prefix: foo_
+object_tag_name: user-tag
+object_tag_value: get-obj-tagging
```
Comment (kaustubh-d) [inline s3/replication/common/tests/system/config/config.yaml:8]: may be use `user-tag-value` as value part.

**Review by kaustubh-d — APPROVED** [review]
one minor comment and one codacy error, please fix before merge.

File: s3/replication/common/tests/system/config/config.yaml (hunk 3 above)
Comment (priyanka25081999) [reply s3/replication/common/tests/system/config/config.yaml]: done

**Review by kaustubh-d — APPROVED** [review]

File: s3/replication/common/tests/system/test_put_object_tagging.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    # Put tagset on the object.
+    command = str(
+        'aws s3api put-object-tagging --bucket ' +
+        bucket_name +
+        ' --key ' +
+        object_name +
+        ' --tagging \'{\"TagSet\" : [{\"Key\":\"user-tag\",'
+        '\"Value\":\"user-tag-value\"}]}\'')
+
+    os.popen(command)
```
Comment (rajkumarpatel2602) [inline s3/replication/common/tests/system/test_put_object_tagging.py:41]: Exit status can be checked and act accordingly. 
```
  exit_status = os.system(command)

    if exit_status == 0:
        print("\nput-bucket-replication passed! ")
    else:
        os.system('rm -rf temp_policy.json')
        print("\nput-bucket-replication failed! ")
        os._exit(exit_status)
```

File: s3/replication/common/tests/system/test_get_object_tagging.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    object_name = config.object_name_prefix + "test"
+    request_id = "dummy-request-id"
+
+    tag_object = S3AsyncGetObjectTagging(session, request_id,
+                                         bucket_name,
+                                         object_name)
+
+    await tag_object.fetch()
+
+    # Validate if tags value matches to default object tag value
```
Comment (rajkumarpatel2602) [inline s3/replication/common/tests/system/test_get_object_tagging.py:60]: replace : "default object tag value" to "object tag  value in config"

File: s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+
+        except aiohttp.client_exceptions.ClientConnectorError as e:
+            self.remote_down = True
+            self._state = S3RequestState.FAILED
+            self._logger.error(fmt_reqid_log(self._request_id) +
+                               "Failed to connect to S3: " + str(e))
+        self._timer.stop()
+        return
+
+    def pause(self):
```
Comment (rajkumarpatel2602) [inline s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py:150]: we will remove these calls in our api class implementations as it won't serve any purpose.

File: s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+        """Returns tags count."""
+        return len(self._response_tags_dict)
+
+    def get_tags_dict(self):
+        """Returns tags dictionary."""
+        return self._response_tags_dict
+
+    def get_tags_value(self, key):
+        """Returns the value for the given key."""
+        return self._response_tags_dict[key]
```
Comment (huanghua78) [inline s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py:66]: If the key does not exist, what will happen?

File: s3/replication/common/tests/system/test_put_object_tagging.py (hunk 4 above)
Comment (priyanka25081999) [reply s3/replication/common/tests/system/test_put_object_tagging.py:41]: done

File: s3/replication/common/tests/system/test_get_object_tagging.py (hunk 5 above)
Comment (priyanka25081999) [reply s3/replication/common/tests/system/test_get_object_tagging.py]: done

File: s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py (hunk 6 above)
Comment (priyanka25081999) [reply s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py]: done

File: s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py (hunk 7 above)
Comment (priyanka25081999) [reply s3/replication/common/src/s3replicationcommon/s3_get_object_tagging.py]: code updated.
done.

**Review by mbcortx — APPROVED** [review]

**Review by rajkumarpatel2602 — APPROVED** [review]

**Comment by rajkumarpatel2602** [comment]
Jenkins Build also passed : http://eos-jenkins.colo.seagate.com/job/Multisite/job/Run-Tests/10/console

**Pull request merged**